src/core/opentelemetry.ml

```diff
@@ -1133,13 +1139,16 @@ module Trace = struct
     (* called once we're done, to emit a span *)
     let finally res =
       let status =
-        match res with
-        | Ok () -> default_status ~code:Status_code_ok ()
-        | Error (e, bt) ->
-          (* add backtrace *)
-          Scope.record_exception scope e bt;
-          default_status ~code:Status_code_error ~message:(Printexc.to_string e)
-            ()
+        match scope.span_status with
+        | Some status -> status
+        | None ->
+          (match res with
+          | Ok () -> default_status ~code:Status_code_ok ()
+          | Error (e, bt) ->
+            (* add backtrace *)
+            Scope.record_exception scope e bt;
+            default_status ~code:Status_code_error
+              ~message:(Printexc.to_string e) ())
```

Comment:
I wonder if the order of preference is correct here 🤔 My thought was that I don't want to override what the user has set.

But then it means that if the user has explicitly set an `Ok` status and an exception is raised, the span will remain in the `Ok` status.

Reply:
looking at what the official sdk does, I found [this](https://opentelemetry.io/docs/languages/go/instrumentation/#set-span-status). It seems like we shouldn't set `Ok` unless it's explicitly done by the user; and so it makes sense to not override the user's explicit choice (maybe the exception is not an actual error).

Reply:
> It seems like we shouldn't set Ok unless it's explicitly done by the user

I agree and adapted the code